src/acl_hal_mmd.cpp

```diff
@@ -1876,6 +1876,14 @@ void acl_hal_mmd_copy_globalmem_to_globalmem(cl_event event, const void *src,
 void acl_hal_mmd_launch_kernel(unsigned int physical_device_id,
                                acl_kernel_invocation_wrapper_t *wrapper) {
   acl_assert_locked();
+
+  const int handle = device_info[physical_device_id].handle;
+  const auto *const mmd = device_info[physical_device_id].mmd_dispatch;
+
+  if (auto *const pfn = mmd->aocl_mmd_simulation_streaming_kernel_info; pfn) {
+    pfn(handle, wrapper->stream_info);
```

Comment:
Klocwork fails to parse C++17 `if` with initializer :disappointed: 
```
174 (Local) /__w/fpga-runtime-for-opencl/fpga-runtime-for-opencl/src/acl_hal_mmd.cpp:1884 UNINIT.STACK.MUST (1:Critical) Analyze
'pfn' is used uninitialized in this function.
  * acl_hal_mmd.cpp:1883: 'pfn' is declared.
  * acl_hal_mmd.cpp:1884: 'pfn' is used, but is uninitialized.
Current status 'Analyze'
```

Reply:
Same issue with Klocwork 22.1.0.69
```
175 (Local) /home/klocwork/fpga-runtime-for-opencl/src/acl_hal_mmd.cpp:1884 UNINIT.STACK.MUST (1:Critical) Analyze
'pfn' is used uninitialized in this function.
  * acl_hal_mmd.cpp:1883: 'pfn' is declared.
  * acl_hal_mmd.cpp:1884: 'pfn' is used, but is uninitialized.
Current status 'Analyze'
```

Reply:
Klocwork passes with the work-around in commit https://github.com/intel/fpga-runtime-for-opencl/pull/103/commits/72be30c192e1f69da4d5ce21f6b3809178bd749b, which means we won't be able to use C++17 `if` initializers for the foreseeable future.